lib/router/transition-intent/named-transition-intent.ts

```diff
@@ -140,6 +140,9 @@ export default class NamedTransitionIntent<T extends Route> extends TransitionIn
     }
 
     merge(newState.queryParams, this.queryParams || {});
+    if (isIntermediate) {
+      merge(newState.queryParams, oldState.queryParams);
+    }
```

Comment:
I think you'll have to either default like this:

```js
    if (isIntermediate) {
      merge(newState.queryParams, oldState.queryParams || {});
    }
```

Or only do it if there are QPs (seems vaguely preferable):

```suggestion
    if (isIntermediate && oldState.queryParams) {
      merge(newState.queryParams, oldState.queryParams);
    }
```